src/dataflow/source/kafka.rs

```diff
@@ -456,11 +454,10 @@ fn downgrade_capability(
     }
     //  Next, we determine the maximum timestamp that is fully closed. This corresponds to the minimum
     //  timestamp across partitions. This value is stored in next_partition_ts
-    for next_ts in next_partition_ts.values() {
-        if *next_ts < min {
-            min = *next_ts
-        }
-    }
+    let min = *next_partition_ts
```

Comment:
This change is correct because
1) per-partition, we only add offsets in increasing order (as verified by the assert statement in server.rs)
2) when we add a new partition, we require it to have a timestamp that is strictly greater than all prior closed timestamps (this invariant is maintained by is_valid_ts in timestamp.rs). We fast forward the stream to achieve that result.

Hence min is guaranteed to increase every time we call .values().min().expect()

It might be worthwhile adding those comments explicitly?